### ISSUE TITLE
[FEA]: Support `ControlMessage` data types for inference stages

### ISSUE THREAD

**Issue by mdemoret-nv** [nv-morpheus/Morpheus#1607]
### Is this a new feature, an improvement, or a change to existing functionality?

New Feature

### How would you describe the priority of this feature request

High

### Please provide a clear description of problem this feature solves

Currently, all of our inference stages only support using the `MultiMessage` classes and their children. As we update the Morpheus stages to utilize `ControlMessage`, the inference stages need to be upgraded as well.

### Describe your ideal solution

Ideally, the inference stages would support both `MultiMessage` and `ControlMessage` in a single class while we deprecate `MultiMessage`. However, `ControlMessage` objects have the additional task settings which area associated with each message. This would be very useful to allow adding an "inference" task to each message which controls properties about the inference on a message by message basis (for example, which model to use). It may make more sense to create the inference stage for `ControlMessage` as a new class to allow supporting using tasks.

## Completion Criteria

- [ ] Determine if the existing class should be updated or a new class should be added specifically for `ControlMessage` types
- [ ] Update any batching and logits calculations to support `ControlMessage`
- [ ] Add options to the class constructor to configure a fallback task if one is not attached to the message
   - [ ] If a message has the inference task attached, it should always default to using those settings for inference
   - [ ] If a message does not have an inference task, it should use the default task set on the inference stage
   - [ ] If a message does not have an inference task, and no default was supplied by the user, an error will be thrown
- [ ] Update all inference testing to use both `MultiMessage` and `ControlMessage`
- [ ] Update all documentation related to inference stages to reflect the new supported type
- [ ] Compare performance between the two message types to ensure no performance degradation.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

**Comment by mdemoret-nv** [comment]
Superceeded by #1802
